### PR TITLE
Use forked version of Pyrsistent

### PR DESCRIPTION
**Pull request by moshez**
This allows us to use a forked version
of Pyrsistent. Currently, it points
to a CHQ fork that has nothing in it
but a "forking stamp". This should be
updated to a hash that has the speed-ups
we need.

### CODE REVIEW

**Comment by sarum90** [comment]
We have some change detector tests that detected your change!

```
23:36:44 * expected: <InstallApplication(virtualenv=<VirtualEnv(root=FilePath('/opt/flocker'))>, package_uri='https://www.example.com/foo/Bar-1.2.3.whl')> !=
23:36:44   actual:   <InstallApplication(virtualenv=<VirtualEnv(root=FilePath('/opt/flocker'))>, package_uri='pip==8.1.1')>
23:36:44 * expected: <GetPackageVersion(virtualenv=<VirtualEnv(root=FilePath('/opt/flocker'))>, package_name='flocker')> !=
23:36:44   actual:   <InstallApplication(virtualenv=<VirtualEnv(root=FilePath('/opt/flocker'))>, package_uri='-r/flocker/requirements.txt')>
```

File: flocker/control/_persistence.py [hunk 1]
```diff
@@ -173,7 +173,7 @@ def default(self, obj):
             result[_CLASS_MARKER] = obj.__class__.__name__
             return result
         elif isinstance(obj, PClass):
-            result = obj.evolver().data
+            result = obj._to_dict()
```
Comment (sarum90) [inline flocker/control/_persistence.py]: Added.

**Comment by sarum90** [comment]
Just occurred to me, we probably have `pip install .` in our documentation a bunch. http://doc-dev.clusterhq.com/gettinginvolved/infrastructure/packaging.html?highlight=pip for instance.

We should probably add --process-dependency-links to those invocations. in the docs.

File: admin/packaging.py [hunk 2]
```diff
@@ -857,6 +857,10 @@ def omnibus_package_builder(
         steps=(
             InstallVirtualEnv(virtualenv=virtualenv),
             InstallApplication(virtualenv=virtualenv,
+                               package_uri='pip==8.1.1'),
```
Comment (sarum90) [inline admin/packaging.py]: Sorry side thought here.

Any chance that if we just update pip or virtualenv in the docker image, we wouldn't have to do any of this?

**Comment by sarum90** [comment]
Updated the docs / tox.ini / and some I think unused stuff in build.yaml to have `--process-dependency-links`. So that will re-trigger the tests.

I'm going to build an RPM on this branch using buildbot, then spin up a little cluster, install flocker from the branch, and verify that the forked version of pyrsistent is being used.

**Comment by sarum90** [comment]
Actually... I'm just going to try to catch the acceptance tests while they are running.

File: admin/packaging.py (hunk 2 above)
Comment (moshez) [inline admin/packaging.py]: I'd like the logic to be self-contained here.

**Comment by sarum90** [comment]
SSH'd to the acceptance test while it was running.

did:
```
/opt/flocker/bin/python
>>> import pyrsistent
>>> help(pyrsistent)     
```
And the help window indicated that it was the ClusterHQ fork.

That was the main test I wanted to run. Given that passing test and the fact that I find the above code to be maintainable enough, I'm fine with this being merged once the tests have completed, so long as they are green.

**Comment by sarum90** [comment]
I'm going to run the merge-pr script on this so that it will be automatically merged once the tests are green.

**Comment by sarum90** [comment]
This has merge conflicts now... that's probably my fault, sorry.

**Comment by sarum90** [comment]
I'll do the manual merge.

**Pull request merged**